src/utils/VideoChannelUtils.ts

```diff
@@ -14,28 +14,37 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState } from "react";
+import { useState, useMemo, useEffect } from "react";
 import { throttle } from "lodash";
 import { Optional } from "matrix-events-sdk";
+import { logger } from "matrix-js-sdk/src/logger";
 import { IMyDevice } from "matrix-js-sdk/src/client";
 import { CallType } from "matrix-js-sdk/src/webrtc/call";
 import { Room } from "matrix-js-sdk/src/models/room";
 import { RoomStateEvent } from "matrix-js-sdk/src/models/room-state";
 import { RoomMember } from "matrix-js-sdk/src/models/room-member";
 
-import { useTypedEventEmitter } from "../hooks/useEventEmitter";
+import { useEventEmitter, useTypedEventEmitter } from "../hooks/useEventEmitter";
 import WidgetStore, { IApp } from "../stores/WidgetStore";
 import { WidgetType } from "../widgets/WidgetType";
 import WidgetUtils from "./WidgetUtils";
-
-const STUCK_DEVICE_TIMEOUT_MS = 1000 * 60 * 60;
+import VideoChannelStore, { VideoChannelEvent, IJitsiParticipant } from "../stores/VideoChannelStore";
 
 interface IVideoChannelMemberContent {
     // Connected device IDs
     devices: string[];
+    // Time at which this state event should be considered stale
+    expires_ts: number;
 }
 
 export const VIDEO_CHANNEL_MEMBER = "io.element.video.member";
+export const STUCK_DEVICE_TIMEOUT_MS = 1000 * 60 * 60;
```

Comment:
entirely optional, but it helps reduce thinking when looking through the code:
```suggestion
export const STUCK_DEVICE_TIMEOUT_MS = 1000 * 60 * 60; // 1 hour
```